Drop obsolete membership code.

shamil-gadelshin opened the pull request:
### Related issues
- [Drop speculative unused fields from membership](https://github.com/Joystream/joystream/issues/516)
- [Drop InitialMembersBalance membership module configuration](https://github.com/Joystream/joystream/issues/516)

Comments:
- buy_membership extrinsic was changed from `paid_term_id` to get membership price to `MembershipFee` trait constant
- `default_membership_fee` was removed from the genesis config.

shamil-gadelshin committed:
runtime: membership: Remove profile fields.

shamil-gadelshin committed:
runtime: membership: Remove old storage fields.

- Remove PaidMembershipTermsById
- ScreeningAuthority
- Remove default membership fee from genesis
- Add MembershipFee constant

bedeho commented on runtime-modules/membership/src/lib.rs:
```diff
@@ -273,24 +209,23 @@ decl_module! {
 
             // ensure enough free balance to cover terms fees
-            ensure!(T::Currency::can_slash(&who, terms.fee), "not enough balance to buy membership");
+            ensure!(T::Currency::can_slash(&who, fee), "not enough balance to buy membership");
```
Are `can_slash` and `slash` correct to use? I know they were already there, but from the docs it says

`Deducts up to value from the combined balance of who, preferring to deduct from the free balance. This function cannot fail.`

So presumably `can_slash` will also count funds that are locked for some purpose, but we really don't want people to be able to use any locked funds for buying a membership. Only free balance should count, the handbook says

```
Free balance of the signer account exceeds `membership_price`. 
```

shamil-gadelshin replied:
Fixed:
- changed `free_balance` check with `usable_balance` (We don't use other balance purposes such as reservation, so, for us, `usable_balance` is `free_balance` minus locked amount.)
- added a test for the fix

bedeho replied:
Excellent, and thanks for the offline explainer.

bedeho reviewed (commented):
So small, so nice, great job, just one question.

shamil-gadelshin committed:
runtime: membership: Fix review comments.

Change slashing balance from free balance to usable balance in order to disable slashing locked funds.

shamil-gadelshin committed:
runtime: membership: Remove default_paid_membership_fee from genesis.
